ovh/resource_ovh_iploadbalancing_tcp_frontend_test.go

```diff
@@ -0,0 +1,140 @@
+	}
+
+	if len(frontends) == 0 {
+		log.Print("[DEBUG] No frontend to sweep")
+		return nil
+	}
+
+	for _, f := range frontends {
+		frontend := &IpLoadbalancingTcpFrontend{}
+
+		if err := client.Get(fmt.Sprintf("/ipLoadbalancing/%s/tcp/frontend/%d", iplb, f), &frontend); err != nil {
+			return fmt.Errorf("Error calling /ipLoadbalancing/%s/tcp/frontend/%d:\n\t %q", iplb, f, err)
+		}
+
+		if !strings.HasPrefix(frontend.DisplayName, test_prefix) {
+			continue
+		}
+
+		err = resource.Retry(5*time.Minute, func() *resource.RetryError {
+			if err := client.Delete(fmt.Sprintf("/ipLoadbalancing/%s/tcp/frontend/%d", iplb, f), nil); err != nil {
+				return resource.RetryableError(err)
+			}
+			// Successful delete
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	err = resource.Retry(5*time.Minute, func() *resource.RetryError {
+		if err := iploadbalancingRefresh(client, iplb, ""); err != nil {
+			return resource.RetryableError(err)
+		}
+		// Successful refresh
+		return nil
+	})
+
+	return nil
+}
+
+func TestAccOvhIpLoadbalancingTcpFrontend_basic(t *testing.T) {
+	iplb := os.Getenv("OVH_IPLB_SERVICE")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: fmt.Sprintf(testAccCheckOvhIpLoadbalancingTcpFrontendConfig_basic, iplb, test_prefix),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"ovh_iploadbalancing_tcp_frontend.testfrontend", "display_name", test_prefix),
+				),
+			},
+		},
+	})
+}
+
+func TestAccOvhIpLoadbalancingTcpFrontend_withfarm(t *testing.T) {
+	iplb := os.Getenv("OVH_IPLB_SERVICE")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: fmt.Sprintf(testAccCheckOvhIpLoadbalancingTcpFrontendConfig_withfarm, iplb, test_prefix, test_prefix),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"ovh_iploadbalancing_tcp_frontend.testfrontend", "display_name", test_prefix),
+				),
+			},
+		},
+	})
+}
+
+const testAccCheckOvhIpLoadbalancingTcpFrontendConfig_basic = `
+resource "ovh_iploadbalancing_tcp_frontend" "testfrontend" {
+   service_name = "%s"
+   display_name = "%s"
+   zone = "all"
+   port = "80,443"
+}
+`
+
+const testAccCheckOvhIpLoadbalancingTcpFrontendConfig_withfarm = `
+data "ovh_iploadbalancing" "iplb" {
+  service_name = "%s"
+}
+
+resource "ovh_iploadbalancing_tcp_farm" "farm" {
+   service_name = "${data.ovh_iploadbalancing.iplb.service_name}"
+   display_name = "%s"
+   zone = "all"
+   port = 80
+}
+
+resource "ovh_iploadbalancing_tcp_frontend" "testfrontend" {
+   service_name = "${data.ovh_iploadbalancing.iplb.service_name}"
+   display_name = "%s"
+   zone = "all"
+   port = "80,443"
+   default_farm_id = "${ovh_iploadbalancing_tcp_farm.farm.id}"
+}
+`
```

Comment:
as mentioned for struct, the same applies here. Consider testing setting some bool value to true and then updating it to false and verifying if it correctly applied on the ovh api.

Reply:
done